spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -135,16 +138,21 @@ def __init__(
                 per_timestep_variables, start=region_id + 1):
             self.__region_ids[variable] = ts_region_id
 
-    def _count_recording_per_slice(self, variable, vertex_slice):
+    def _count_recording_per_slice(
+            self, variable, vertex_slice, max_rewires_per_ts):
```

Comment:
I think it makes more sense to do the calculation for this outside of this function.  If this is not passed in as a "variable" in the constructor but instead stored as max_rewires_per_ts in an instance variable, it will then not end up in a loop that calls this function, and can be dealt with separately.